homeassistant/components/frontend/storage.py

```diff
@@ -0,0 +1,79 @@
+"""API for persistent storage for the frontend."""
+from functools import wraps
+import voluptuous as vol
+
+from homeassistant.components import websocket_api
+
+DATA_STORAGE = 'frontend_storage'
+STORAGE_VERSION_USER_DATA = 1
+STORAGE_KEY_USER_DATA = 'frontend.user_data_{}'
+
+
+async def async_setup_frontend_storage(hass):
+    """Set up frontend storage."""
+    hass.data[DATA_STORAGE] = {}
+    hass.components.websocket_api.async_register_command(
+        websocket_set_user_data
+    )
+    hass.components.websocket_api.async_register_command(
+        websocket_get_user_data
+    )
+
+
+def with_store(orig_func):
```

Comment:
There are two usage of `with_store`, in `set` function, both `store` and `data` are used in function. But in `get` function, only `data` is used. So, I suggest:

This decorate should be named as
```python
def with_user_data(with_store: boolean=False):
    def wrap_func(orig_func):
        @wraps(orig_func)
        async def with_store_func(hass, connection, msg):
            ...
            if with_store:
                await orig_func(..., store, data[user_id])
            else:
                await orig_func(..., data[user_id])
        return with_store_func
    return wrap_func
```

Reply:
The store is still used inside the decorator, regardless if the wrapped function uses it. It is used to load data if none cached yet.

Since it's two functions, I don't mind passing 1 extra arg, to avoid extra complexity.